test_chess.py

```diff
@@ -0,0 +1,156 @@
+import pytest
+
+import ai_engine
+from unittest.mock import Mock
+from Piece import Knight, King, Rook
+from enums import Player, GameStatus
+import chess_engine
+
+"""
+Unit tests
+"""
+
+
+@pytest.fixture
+def game_state_mock():
+    return Mock()
+
+
+# Tests for the function get_valid_peaceful_moves in class Knight
+
+def test_knight_middle_empty(game_state_mock):
+    """
+    The knight is in the middle, all the surrounding squares are empty
+    """
+    game_state_mock.get_piece = Mock(return_value=Player.EMPTY)
+    white_knight = Knight('n', 3, 4, Player.PLAYER_1)
+    result = white_knight.get_valid_peaceful_moves(game_state_mock)
+    assert set(result) == {(1, 3), (1, 5), (2, 2), (2, 6), (4, 2), (4, 6), (5, 3), (5, 5)}
+
+
+def test_knight_right_empty(game_state_mock):
+    """
+    The knight is on the left side of the board, half of the squares are empty (from the right side)
+    and the rest are off the board
+    """
+    game_state_mock.get_piece = Mock(return_value=Player.EMPTY)
+    white_knight = Knight('n', 2, 0, Player.PLAYER_1)
+    result = white_knight.get_valid_peaceful_moves(game_state_mock)
+    result = [i for i in result if 0 <= i[0] < 8 and 0 <= i[1] < 8]
+    assert set(result) == {(0, 1), (1, 2), (3, 2), (4, 1)}
+
+
+def test_knight_upper_empty(game_state_mock):
+    """
+    The knight is on the bottom of the board, half of the squares off the board, one of the squares contain an enemy
+    piece and the rest are empty
+    """
+    game_state_mock.get_piece.side_effect = lambda row, col: Knight('n', row, col, Player.PLAYER_1) \
```

Comment:
nice but a bit unreadable - maybe you can create a specific function with some parameters that does the same your lambda function does.